### PR TITLE
carla: init at 2.0.0

### PR DESCRIPTION
**Pull request by minijackson**
###### Motivation for this change

To have more choices in audio production software. Carla seems fairly used by people using the Jack ecosystem. #11190 mentions Carla, so @Pitometsu, @joelmo and @magnetophon might be interested.

Tested (a bit) from the release-19.03 branch because I don't have the resources of compiling QT from scratch ^^

If someone with a more complete audio setup wants to test it, I'd be happy :D 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by magnetophon** [comment]
Thanks for making this!
I cherrypicked the commit on unstable, and when I ran ``carla`` I got:
```
  File "/nix/store/146pjlvrf3rwccpkbf5qkpc1mgh0laqb-carla-2.0.0/share/carla/carla", line 2
    export PATH='/nix/store/kimimvhimclyfzlncpg36zjni3wn70nq-python3-3.7.3/bin:/nix/store/a34f16h8ih6dvvr11fwfjbij0vck8zyq-python3.7-rdflib-4.2.2/bin:/nix/store/rpayih9fb4ms2wjvmyn6kp7kmsizim49-python3.7-setuptools-40.8.0/bin:/nix/store/p351a2bc6zvi4vicdrjwvj4ypls1cfsq-python3.7-pyliblo-0.10.0/bin:/nix/store/rlx99nx951mb919x96viv6hfr5hxa4l0-python3.7-PyQt-5.11.3/bin'${PATH:+':'}$PATH
              ^
SyntaxError: invalid syntax
```

**Comment by FRidh** [comment]
@GrahamcOfBorg eval

**Comment by minijackson** [comment]
Welp, I must have missed something or tested the wrong result, sorry about that :-/ It should be fixed, now

**Comment by magnetophon** [comment]
All working now; great!
I'll test some more tonight.

File: pkgs/applications/audio/carla/default.nix [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+  buildInputs = [
+    file liblo alsaLib fluidsynth ffmpeg jack2 libpulseaudio libsndfile
+  ] ++ pythonPath
+    ++ optional withQt qtbase
+    ++ optional withGtk2 gtk2
+    ++ optional withGtk3 gtk3;
+
+  installPhase = ''
+    make PREFIX=$out install
```
Comment (veprbl) [inline pkgs/applications/audio/carla/default.nix]:
```
installFlags = [ "PREFIX=$(out)" ];
```

File: pkgs/applications/audio/carla/default.nix [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  meta = with stdenv.lib; {
+    homepage = http://kxstudio.sf.net/carla;
+    description = "An audio plugin host";
+    longDescription = ''
+      It currently supports LADSPA (including LRDF), DSSI, LV2, VST2/3
+      and AU plugin formats, plus GIG, SF2 and SFZ file support.
+      It uses JACK as the default and preferred audio driver but also
+      supports native drivers like ALSA, DirectSound or CoreAudio.
+    '';
+    license = licenses.gpl2;
```
Comment (veprbl) [inline pkgs/applications/audio/carla/default.nix]:
```suggestion
    license = licenses.gpl2Plus;
```

**Comment by minijackson** [comment]
Thanks for the review @veprbl !

**Comment by magnetophon** [comment]
Works just fine here, thanks!

**Comment by veprbl** [comment]
@GrahamcOfBorg build carla

File: pkgs/applications/audio/carla/default.nix [hunk 3]
```diff
@@ -0,0 +1,71 @@
+{ stdenv, fetchFromGitHub, alsaLib, file, fluidsynth, ffmpeg, fftw, jack2,
+  liblo, libpulseaudio, libsndfile, makeWrapper, pkgconfig, python3,
+  python3Packages, which, withFrontend ? true,
```
Comment (FRidh) [inline pkgs/applications/audio/carla/default.nix]: keep either python3 or python3Packages

**Comment by minijackson** [comment]
Thanks for the pointer @FRidh 

Also added `which` to the PATH, else I couldn't get it to run in a `--pure` nix shell

**Comment by veprbl** [comment]
@GrahamcOfBorg build carla

**Pull request merged**